src/openforms/utils/json_logic.py

```diff
@@ -49,3 +49,123 @@ def normalize_value(value: Union[dict, list, tuple, int, str]) -> JSONLogicValue
+
+    """returns the value inside a JsonLogicTest"""
+    if isinstance(value, JsonLogicTest) and value.operator == "date":
+        return find_json_logic_test_value(value.values[0])
+    if isinstance(value, JsonLogicTest) and value.operator == "var":
+        return value.values[0]
+
+    else:
+        return value
+
+
+def convert_json_logic_in_json_tree(
+    logics, keys_mapped_to_component, keys_mapped_to_form, updated_submission_data
+):
+
+    """returns a dict (potentially nested) with complemetentary data regarding the field"""
+
+    if isinstance(logics, JsonLogicTest):
+        logic_dict = {}
+        logic_dict["nodeType"] = "operation"
+        logic_dict["operator"] = logics.operator
+        logic_dict["children"] = []
+        for index, value in enumerate(logics.values):
+
+            # There is an operation
+            if isinstance(value, JsonLogicTest):
+                if value.operator in ["var", "date"]:
+                    key = find_json_logic_test_value(logics.values[index])
+
+                    # leaf not related to component --> just a value compared
+                    if not keys_mapped_to_component.get(key):
+                        logic_dict["value"] = value.values[0]
+
+                    # Leaf is a field
+                    else:
+                        leaf_node = {
+                            "nodeType": "leaf",
+                            "key": key,
+                            "actualValue": updated_submission_data.get(key, ""),
+                            "stepName": keys_mapped_to_form[key].form_definition.name,
+                            "label": keys_mapped_to_component[key].get("label", ""),
+                        }
+                        logic_dict["children"].append(leaf_node)
+                elif value.operator == "rdelta":
+                    logic_dict["value"] = value.values
+
+                # No relevant information with this operator
+                elif value.operator == "today":
+                    pass
+                else:
+                    logic_dict["children"].append(
+                        convert_json_logic_in_json_tree(
+                            value,
+                            keys_mapped_to_component,
+                            keys_mapped_to_form,
+                            updated_submission_data,
+                        )
+                    )
+
+            # No operation, it's a basic value
+            else:
+                logic_dict["value"] = value
+
+        return logic_dict
+
+    # array of rules
+    else:
+        logic_list = []
+        for logic in logics:
+            logic_list.append(
+                convert_json_logic_in_json_tree(
+                    logic,
+                    keys_mapped_to_component,
+                    keys_mapped_to_form,
+                    updated_submission_data,
+                )
+            )
+        return logic_list
```

Comment:
this can probably be a list comprehension

```suggestion
        return [
            convert_json_logic_in_json_tree(
                logic,
                keys_mapped_to_component,
                keys_mapped_to_form,
                updated_submission_data,
            ) for logic in logics
        ]
```